pkg/resource/deploy/source_eval.go

```diff
@@ -562,7 +562,8 @@ func (rm *resmon) Invoke(ctx context.Context, req *pulumirpc.InvokeRequest) (*pu
 
 	args, err := plugin.UnmarshalProperties(
 		req.GetArgs(), plugin.MarshalOptions{
-			Label:         label,
+			Label: label,
+			// RejectUnknowns: true  //  should we just fail fast here if we see unknowns?
```

Comment:
I'd expect to reject unknowns here but I don't have a holistic picture of this option, unfotunately

Reply:
This option simply scans the data structure and if it finds any unknowns it throws an exception. Something like this:

```
if hasUnknowns(props) {
   return fmt.Errorf("Unexpected unknown property")
} else {
   return nil
}
```

@mikhailshilkov that sounded like a decent option to me also, but now it seems that with my other fix to Marshal/Unmarshal functions landing on master, I cannot repro the bug... That is, unknowns get property filtered out and do not cause panic. Moreover, when Pulumi resolves them to known values, the example works - and passing Output to RPC function works, contrary to wha the team believes. I'm therefore hesitating to put this fix in, as it would break seemingly working examples.